src/components/viewPartsButton.jsx

```diff
@@ -16,23 +16,69 @@ const ViewParts = () => {
       });
   };
 
+  const downloadFile = (part_id, fileName) => {
+    fetch(`/api/v0/parts/${part_id}/file`)
+      .then((res) => res.blob())
+      .then((blob) => {
+        const a = document.createElement("a");
+        const href = URL.createObjectURL(blob);
+        a.setAttribute("download", fileName);
+        document.body.appendChild(a); // apparently, this line is needed to make 'a.click()' work in firefox/ensures that the link is part of the document
+        a.href = href;
+        a.click(); // Trigger the download
+        document.body.removeChild(a); // Clean up by removing the link from the document
+        URL.revokeObjectURL(href); // Free up memory by revoking the object URL
```

Comment:
These lines are kind of complicated to read through so they might be good to extract into a new function like downloadBlobToLocalMachine or something and put all the lines into that so that it reads easier in here